integration/import/package.json

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
where's package.json for esm?

Reply:
I'm forcing esm with the `.mjs` extension in these tests.

Reply:
I think I'm bit confused. Purpose of this integration test is trying to mimc actual user's package.json with various config, so we can't enforce something since we can't ask user to adjust package.json. That's reason there was `cjs/package.json`, and I expect esm have same package.json have `type:module` to test esm import when user decides to go with native esm. So in total, we expect at least 3 tests - `cjs` without any `type` (for node.js doesn't support `type` property) / `cjs` with `type:cjs` / esm with `type:module`. 

User can choose either use mjs, or decide `type:module` and treat all js as esm.

Reply:
Mind elaborate how current test covers above cases?

Reply:
My thought was that by invoking cjs and mjs explicitly with file extensions, I'm exacting the codepaths in node that load the code with the correct loader. Whether that loader option is invoked via a package.json or a type field is a node implementation detail that may expand in the future. IMHO adding a separate package.json is just another way node gives you to chose a loader but I don't think it needs to be a focus of these tests.

Right now the assertions being made ensure asserting that the module exports are equivalent when run via cjs and esm.

Reply:
> IMHO adding a separate package.json is just another way node gives you to chose a loader but I don't think it needs to be a focus of these tests.

as long as your test can represent real user's package.json usecase to test, it should be ok. But I'm bit hard to understand if it's covered. So current way can verify flows like

1. user uses old / latest node.js, package.json doesn't have type field and want to use cjs
2. user uses latest node.js, package.json have type field to cjs, want to use cjs
3. user uses latest node.js package.json have type field to esm, want to use esm
4. (need to be added) webpack / other bundle picks up esm correctly

> Whether that loader option is invoked via a package.json or a type field is a node implementation detail that may expand in the future

This isn't about node.js internal, As stated this test try to mimic current real world package.json structures as user load their module in various different way. Regardless of how node.js treats, we should support existing ergonomics user uses.

Reply:
For sure, For the flows you outlined - the answers are declared in the package.json

1) `main` field is discovered -> dist/cjs is loaded
2 & 3) `exports` field is discovered -> import ? -> dist/esm : dist/cjs
4) Webpack implements resolution logic based on [these fields](https://github.com/webpack/enhanced-resolve#resolver-options) as well

I don't know if there is value to adding more tests here since the existing tests now validate that both esm and cjs can be requested and contain a consistent API.